### PR TITLE
New nuget release prep

### PR DESCRIPTION
**Pull request by harleenkohli**

### CODE REVIEW

File: src/build.props [hunk 1]
```diff
@@ -12,6 +12,9 @@
     <VersionPrefix>2.1.0</VersionPrefix>
     <VersionSuffix></VersionSuffix>
     <PackageVersionSuffix>-rtm.0</PackageVersionSuffix>
+    <SarifVersion>2.1.0</SarifVersion>
+    <SchemaVersion>2.1.0-rtm.0</SchemaVersion>
+    <SdkVersion>2.1.0</SdkVersion>
```
Comment (harleenkohli) [inline src/build.props]: this is a wip PR to get verbal approval on the approach i am taking here. I have introduced these three variables each for the appropriate purpose of defining:

- SarifVersion - shall remain "2.1.0" to correlate with the spec version.
- SchemaVersion - shall correlate with the schema version on json schema store. current value is [http://json.schemastore.org/sarif-2.1.0-rtm.0](http://json.schemastore.org/sarif-2.1.0-rtm.0)
- SdkVersion - current Semantic version for the SDK. Must be updated before every nuget drop.

File: src/build.props [hunk 2]
```diff
@@ -12,6 +12,9 @@
     <VersionPrefix>2.1.0</VersionPrefix>
     <VersionSuffix></VersionSuffix>
     <PackageVersionSuffix>-rtm.0</PackageVersionSuffix>
```
Comment (harleenkohli) [inline src/build.props]: i am currently removing all dependencies on these values, i have one remaining, but thought of raising the PR first to get eyes on the approach.

File: src/Sarif.Multitool/ValidateCommand.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ namespace Microsoft.CodeAnalysis.Sarif.Multitool
 {
     internal class ValidateCommand : AnalyzeCommandBase<SarifValidationContext, ValidateOptions>
     {
-        public override string Prerelease => VersionConstants.Prerelease;
+        public override string Prerelease => string.Empty;
```
Comment (harleenkohli) [inline src/Sarif.Multitool/ValidateCommand.cs]: >Prerelease => string.Empty; [](start = 31, length = 27)

I am unsure about the purpose of the "prerelease" string, for now i have updated the value from the actual build variable (which was an empty string) to surgically break the dependency. Should we simply remove this variable usage now?

File: src/build.props (hunk 2 above)
Comment (michaelcfanning) [inline src/build.props]: delete unused things

File: src/build.props (hunk 1 above)
Comment (michaelcfanning) [inline src/build.props]: >SdkVersion [](start = 5, length = 10)

Maybe use PackageVersion for this?

File: src/build.props [hunk 4]
```diff
@@ -12,6 +12,9 @@
...
+    <SarifVersion>2.1.0</SarifVersion>
```
Comment (michaelcfanning) [inline src/build.props]: >SarifVersion [](start = 5, length = 12)

Call this StableSarifVersion

File: src/build.props [hunk 5]
```diff
@@ -12,6 +12,9 @@
...
+    <SchemaVersion>2.1.0-rtm.0</SchemaVersion>
```
Comment (michaelcfanning) [inline src/build.props]: >SchemaVersion [](start = 5, length = 13)

SchemaVersionAsPublishedToSchemaStoreOrg

File: src/build.props (hunk 1 above)
Comment (michaelcfanning) [reply src/build.props]: Add comments to this file that explain the purpose of all these things.

---
In reply to: [282281303](https://github.com/microsoft/sarif-sdk/pull/1484#discussion_r282281303) [](ancestors = 282281303)

File: src/ReleaseHistory.md [hunk 6]
```diff
@@ -374,3 +374,4 @@
 * API BREAKING: Add regex patterns for guid and language in schema.
 * API NON-BREAKING: Add `run.specialLocations` in schema. https://github.com/oasis-tcs/sarif-spec/issues/396
 * API BREAKING: Improve `address` object design. https://github.com/oasis-tcs/sarif-spec/issues/401
+* API NON-BREAKING: `PhysicalLocation.id` property is getting lost during 2.1.0 pre-release transformation. https://github.com/microsoft/sarif-sdk/issues/1479
```
Comment (michaelcfanning) [inline src/ReleaseHistory.md]: >API NON-BREAKING:  [](start = 2, length = 18)

add a new version here

**Comment by michaelcfanning** [comment]
>         public static string SemanticVersion

StableSarifVersion

---
Refers to: src/Sarif/SarifUtilities.cs:39 in 2a7c1b0. [](commit_id = 2a7c1b0297604ab61814654cea89c8e1c94256d5, deletion_comment = False)

File: src/Sarif.Multitool/ExportRulesMetadataCommand.cs [hunk 7]
```diff
@@ -19,6 +19,6 @@ public override IEnumerable<Assembly> DefaultPlugInAssemblies
             }
         }
 
-        public override string Prerelease { get { return VersionConstants.Prerelease; } }
+        public override string Prerelease { get { return string.Empty; } }
```
Comment (michaelcfanning) [inline src/Sarif.Multitool/ExportRulesMetadataCommand.cs]: >Prerelease [](start = 31, length = 10)

i don't think we need this!

File: src/Sarif.Multitool/ValidateCommand.cs (hunk 3 above)
Comment (michaelcfanning) [reply src/Sarif.Multitool/ValidateCommand.cs]: yes, you're right, remove it.

---
In reply to: [282281902](https://github.com/microsoft/sarif-sdk/pull/1484#discussion_r282281902) [](ancestors = 282281902)

**Comment by harleenkohli** [comment]
@lgolding is added to the review. #Closed

File: src/build.props [hunk 8]
```diff
@@ -9,9 +9,15 @@
     <Company Condition=" '$(Company)' == '' ">Microsoft</Company>
     <Product Condition=" '$(Product)' == '' ">Microsoft SARIF SDK</Product>
     <Copyright Condition=" '$(Copyright)' == '' ">© Microsoft Corporation. All rights reserved.</Copyright>
+    
+    <!-- VersionPrefix denotes the stable Version of SARIF Specifications. -->
```
Comment (michaelcfanning) [inline src/build.props]: >denotes the stable Version of SARIF Specification [](start = 23, length = 49)

No. VersionPrefix denotes the assembly version that we will version out of band with the SARIF specification version.

**Pull request merged**